pkg/federation/sdlmerge/interface_type_extending.go

```diff
@@ -26,23 +29,28 @@ func (e *extendInterfaceTypeDefinitionVisitor) EnterDocument(operation, _ *ast.D
 }
 
 func (e *extendInterfaceTypeDefinitionVisitor) EnterInterfaceTypeExtension(ref int) {
-	nodes, exists := e.document.Index.NodesByNameBytes(e.document.InterfaceTypeExtensionNameBytes(ref))
+	nameBytes := e.document.InterfaceTypeExtensionNameBytes(ref)
+	nodes, exists := e.document.Index.NodesByNameBytes(nameBytes)
 	if !exists {
 		return
 	}
-
 	hasExtended := false
+	isEntity := false
 	for _, node := range nodes {
 		if node.Kind != ast.NodeKindInterfaceTypeDefinition {
 			continue
 		}
 		if hasExtended {
-			e.Walker.StopWithExternalErr(operationreport.ErrSharedTypesMustNotBeExtended(e.document.InterfaceTypeExtensionNameString(ref)))
+			e.Walker.StopWithExternalErr(*getMultipleExtensionError(isEntity, nameBytes))
```

Comment:
Missing `return` or `break` statement?

Reply:
No; we need to check that we're not trying to extend more than once.

Reply:
do we need to have multiple external errors for all nodes?

Burak means that after having an external error we don't have to proceed with the remaining code

As usually after calling StopWithExternal error we interrupt function execution

Reply:
The code ceases immediately after it hits this, so I am not sure what you mean?

Reply:
we stop the walker, but not the `for` loop

stopping walker means that it stops iterating root nodes and calling enter callbacks

Reply:
I misunderstood how that worked. I have added returns to all cases.